tools/schemapi/utils.py

```diff
@@ -22,6 +28,133 @@
 }
 
 
+class _TypeAliasTracer:
```

Comment:
Thinking this will be the main topic for review, due to its contribution to the diff.

The obvious choices for a postfix on the `TypeAlias` name would create some odd results. In [this selection](https://github.com/dangotbanned/altair/blob/d0c01eb6259c4c79cce42add2bcc8bb3d1ec41ed/altair/vegalite/v5/schema/_typing.py#L502-L546) you can see some examples that make it difficult to apply a common postfix, without causing a collision. 

Below I've included 4 others which were an issue. 
An alternative argument to `_TypeAliasTracer(fmt=...)` would need to make sense across all of these: 

| Input | Output |
|--------|--------|
| `MarkType` | `MarkType_T` |
| `Mark` | `Mark_T` |
| `RangeEnum` | `RangeEnum_T`  |
| `ProjectionType` | `ProjectionType_T` |
| `ResolveMode` | `ResolveMode_T` | 
| `ScaleInterpolateEnum` | `ScaleInterpolateEnum_T` | 
| `ScaleType` | `ScaleType_T` | 
| `SelectionResolution` | `SelectionResolution_T` | 
| `SelectionType` | `SelectionType_T` | 
| `TypeForShape` | `TypeForShape_T` | 
| `Type` | `Type_T` |